File: solutions/java/etl/1/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Etl {
    Map<String, Integer> transform(Map<Integer, List<String>> old) {
        Map<String, Integer> result = new HashMap<>();
        old.forEach((key, value) -> value.forEach(letter -> result.put(letter.toLowerCase(), key)));
        
        return result;
    }
}

File: solutions/cpp/etl/1/etl.cpp
#include "etl.h"
#include <cctype>  // for std::tolower

namespace etl {

std::map<char, int> transform(const std::map<int, std::vector<char>>& old) {
    std::map<char, int> result;

    for (const auto& pair : old) {
        for (char c : pair.second) {
            result[std::tolower(c)] = pair.first;
        }
    }

    return result;
}

}